exonum-java-binding/core/rust/ejb-app/src/node_builder.rs

```diff
@@ -41,116 +105,101 @@ fn service_factories() -> HashMap<String, Box<ServiceFactory>> {
         Box::new(TimeServiceFactory) as Box<ServiceFactory>,
     );
-    service_factories.insert(
-        EJB_SERVICE.to_owned(),
-        Box::new(JavaServiceFactory) as Box<ServiceFactory>,
-    );
     service_factories
 }
 
-#[doc(hidden)]
-pub fn services_to_enable<P: AsRef<Path>>(path: P) -> HashSet<String> {
-    // Return default list if config file not found.
-    let mut services = load_enabled_services(path).unwrap_or_else(|_| {
-        let mut services = HashSet::new();
-        services.insert(CONFIGURATION_SERVICE.to_owned());
-        services
-    });
-
-    // Add EJB_SERVICE if it's missing
-    services.insert(EJB_SERVICE.to_owned());
-
-    services
-}
-
-pub fn create() -> fabric::NodeBuilder {
-    let services = services_to_enable(PATH_TO_SERVICES_TO_ENABLE);
-    let mut service_factories = service_factories();
-
-    let mut builder = fabric::NodeBuilder::new();
-    for service_name in &services {
-        match service_factories.remove(service_name) {
-            Some(factory) => {
-                builder = builder.with_service(factory);
-            }
-            None => panic!("Found unknown service name {}", service_name),
-        }
-    }
-    builder
-}
-
 #[cfg(test)]
 mod tests {
     use super::*;
     use std::io::Write;
     use tempfile::{Builder, TempPath};
 
-    fn create_config(filename: &str, cfg: &str) -> TempPath {
-        let mut cfg_file = Builder::new().prefix(filename).tempfile().unwrap();
-        writeln!(cfg_file, "{}", cfg).unwrap();
-        cfg_file.into_temp_path()
-    }
-
     #[test]
-    fn no_config() {
-        let services_to_enable = services_to_enable("");
-        assert_eq!(services_to_enable.len(), 2);
-        assert!(services_to_enable.contains(EJB_SERVICE));
-        assert!(services_to_enable.contains(CONFIGURATION_SERVICE));
+    #[should_panic(expected = "Unable to load services definition")]
+    fn nonexistent_file() {
+        prepare_service_factories("nonexistent");
     }
 
     #[test]
-    fn empty_list() {
-        let cfg = create_config("empty_list.toml", "services = []");
-        let services_to_enable = services_to_enable(cfg);
-        assert_eq!(services_to_enable.len(), 1);
-        assert!(services_to_enable.contains(EJB_SERVICE));
+    #[should_panic(expected = "At least one user service should be defined")]
+    fn missing_user_services() {
+        let cfg = create_config(
+            "no_user_services.toml",
+            r#"
+                system_services = ["configuration", "btc_anchoring", "exonum_time"]
+                [user_services]
+             "#,
+        );
+        prepare_service_factories(cfg);
     }
 
     #[test]
-    fn duplicated() {
+    #[should_panic(expected = "Unknown system service name")]
+    fn unknown_system_service() {
         let cfg = create_config(
-            "duplicated.toml",
-            "services = [\"btc-anchoring\", \"btc-anchoring\"]",
+            "no_user_services.toml",
```

Comment:
Here and below, the file name is misleading. Why do we pass it *at all* if the method returns a path (to use the file name assertions :thinking:)?

Reply:
Actually, I agree with you and in the branch with extended integration tests I removed the name parameter. Going to remove it here as well.

Reply:
Well, as I mentioned, it _may_ be used in assertions on error messages, but it isn't now, so better be removed.

Reply:
Done. For the moment (until dynamic services) I maintain two copies of testing helper `create_config` in `core` and `ejb-app` because there is no way to share the common testing code for testing configuration only unless I create separate crate like "test_utils". Which looks kinda overhead for the moment but would be useful if we wanted such functionality in the integration tests.